Escape quotes in eval(), unit tests, and others

seoj opened the pull request:
Added new function escapeString() in /msl-client-browser/appcontainer-driver.js and modified all calls to eval() to escape the string before concatenating into expression.

Added new karma test under /msl-client-browser/ to unit test this new functionality, along with a readme to explain how to run the unit test.

Also changed /msl-client-browser/package.json where the property "repositories" should have been "repository." This was causing warning messages when npm install was run.

seoj commented:
One thing that is pending is that the new unit tests under /msl-client-browser/ are not run in Travis-CI. Not sure how to do the configuration to run this unit test along with the e2e.

kood1 commented:
We should okay (for now) with the e2e tests that ran successfully in Travis-CI.  Once we merge this in, I will release it out to cdnjs.  Thanks!

kood1 commented:
@seoj - quick question.  Did we add any test for the new change?  Perhaps we could modify one of the existing test to use double quotes.

seoj commented:
@kood1 I added a separate karma test under msl-client-browser directory which tests just the function to escape the quotes. We could probably try using a quote selector in one of the e2e to include this new functionality.

Just thought that including the test ONLY in the e2e wouldn't have been enough to cover all cases for the new function, hence, the new unit test.

kood1 commented:
Yup, agreed on your unit test comment.  Will definitely follow up to work on broader unit test coverage.  Thanks!  Will merge this in soon.

seoj commented:
Updated escapeString() function to escape backslashes along with single and double quotes as per Jake's comment. Updated unit test to test this case too.

kood1 commented:
@seoj - looks like tests failed on Travis... https://travis-ci.org/FINRAOS/MSL/builds/47183857

seoj commented:
Odd. I am able to build and pass all tests in my fork travis [https://travis-ci.org/seoj/MSL/builds/47183855](https://travis-ci.org/seoj/MSL/builds/47183855). The e2e test also pass in my local... 

It seems to be having trouble at line 128:
>var intrReq = JSON.parse(resp).xhr_1;
>expect(intrReq.xhr.url).toBe('/services/getservice?term=testGet');
any ideas?

kood1 commented:
I just reran the build and it passed.  Perhaps a glitch on travis?  Will merge this in now.

kood1 commented:
@seoj - I am making a few changes to get ready for the release of browser client to cdnjs.  Since cdnjs syncs directly with GitHub, it will pickup all js files inside msl-client-browser folder.  In order to only release the necessary js files, I will move the karma config file and jasmine spec file to test/msl-client-browser folder instead.  Since msl-client-browser is not an npm package, I think this is okay.

I will include these tests to be executed with the e2e-script as well since we use this script for our Travis-CI builds.

Thanks for making this code fix by the way! ;)
